Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -66,50 +65,72 @@ Item * createItem(int key, int value, int * success) {
         *success = SUCCESS;
         return item;
     }
+    return NULL;
 }
 
+// creates a new HashTable struct
+// returns new HashTable struct
+HashTable * createHashTable(int size) {
+    if (size < 1 || size > LIMIT) return NULL;
+    HashTable * hashTable = malloc(sizeof(HashTable));
+    hashTable->items = malloc(sizeof(Item) * size);
+    if (hashTable == NULL || hashTable->items == NULL) {
+        return NULL;
+    }
+    hashTable->size = size;
+    for (int i = 0; i < size; i++) {
+        hashTable->items[i] = NULL;
+    }
+    return hashTable;
+}
+
+
 // insert takes in a key and value
 // calls a helper function to create an Item struct with the provided key and value
 // adds the new Item to the hashTable
 // if the expected value is in the hashTable at the expected key then returns a success code
-int insert(int type, int key, int value) {
+int insert(HashTable * hashTable, int key, int value) {
     int success = ERROR;
-    if (type == TEST) {
-        testHash[key] = createItem(key, value, &success);
-        if (testHash[key]->value != value) {
-            return ERROR;
-        }
-        
-    } else if (type == EXPECTED) {
-        expectedHash[key] = createItem(key, value, &success);
-        if (expectedHash[key]->value != value) {
-            return ERROR;
-        }
+    hashTable->items[key] = createItem(key, value, &success);
+    if (hashTable->items[key]->value != value) {
+        return ERROR;
     }
-    return SUCCESS;
+    return success;
 }
 
 // bulkInsert receives an array of keys and an array of values
 // key[i] will be mapped to value[i] when building the hashTable
 // loops through the values to create a key for the current value then inserts it into the existing hashTable.
 // returns a SUCCESS code if all values are added into the hashTable successfully 
-int bulkInsert(int type, int *keys, int *values) {
+int bulkInsert(HashTable * hashTable, int *keys, int *values) {
+    int success = ERROR;
     for (int i = 0; i < SIZE; i++) {
         int key = hashing(keys[i]);
-        return insert(type, key, values[i]);
-   }
+        success = insert(hashTable , key, values[i]);
```

Comment:
`success` is being overwritten without being checked each loop. The returned `success` will be the returned value of the last call to `insert`.
The docs specify that `SUCCESS` is returned if **all** values are added successfully. So may want to add a check here if a insert fails.